packages/frontend/src/pages/bridge-details/BridgeDetailsPage.tsx

```diff
@@ -60,11 +63,25 @@ export const BridgeDetailsPage = () => {
         </IconButton>
       </Box>
 
-      <BridgeDetails bridge={bridge} />
+      <MemoizedBridgeDetails bridge={bridge} />
+
+      <Stack spacing={2}>
+        <Box display="flex" justifyContent="flex-start" alignItems="center">
+          <Typography variant="h6">Devices</Typography>
+          <IconButton onClick={refreshNow}>
+            <Refresh />
+          </IconButton>
+          {timer != null && (
+            <Typography variant="body2" color="textSecondary">
+              Auto-refresh in {(timer / 1000).toFixed(0)} seconds...
+            </Typography>
+          )}
+        </Box>
 
-      {devices && (
-        <DeviceList devices={devices} timer={timer} onRefresh={refreshNow} />
-      )}
+        {devices && (
+          <MemoizedDeviceList devices={devices} onRefresh={refreshNow} />
```

Comment:
As you moved the timer out of the device list, the `onRefresh` prop shouldn‘t be needed anymore.